lalrpop/src/normalize/token_check/mod.rs

```diff
@@ -131,17 +165,49 @@ impl<'grammar> Validator<'grammar> {
             // If there is no extern token definition, then collect
             // the terminal literals ("class", r"[a-z]+") into a set.
             None => match term {
-                TerminalString::Bare(c) => {
-                    // Bare identifiers like `x` can never be resolved
-                    // as terminals unless there is a conversion
-                    // defined for them that indicates they are a
-                    // terminal; otherwise it's just an unresolved
-                    // identifier.
-                    panic!("bare literal `{}` without extern token definition", c);
-                }
-                TerminalString::Literal(l) => {
-                    self.all_literals.entry(l).or_insert(span);
-                }
+                // FIMXE: Should not allow undefined literals if no CatchAll
+                TerminalString::Bare(c) => match self.match_mappings {
+                    Some(ref m) => {
+                        if let Some(v) = m.get(&term) {
+                            // FIXME: I don't think this span here is correct
+                            let vl = v.as_literal().expect("must map to a literal");
+                            self.all_literals.entry(*vl).or_insert(span);
+                        } else {
+                            return_err!(span, "terminal `{}` does not have a match mapping defined for it",
```

Comment:
I would expect that this path is unreachable, right? Because we would have treated any bare literal that has no `match` mapping as an (undefined) nonterminal and errored out earlier.